ENIGMAsystem/SHELL/Platforms/General/PFmain.cpp

```diff
@@ -166,6 +174,35 @@ int updateTimer() {
   return 0;
 }
 
+void wait(int* mutex) {
+  while (*mutex <= 0);
+  (*mutex)--;
+}
+
+void signal(int* mutex) {
+  (*mutex)++;
+}
+
+void fireEventsFromQueue(int* mutex) {
+  // wait(mutex);
+  while (!posted_async_events.empty()) {
+    enigma_user::ds_map_clear(enigma_user::async_load);
+
+    std::map<std::string, variant> event = posted_async_events.front();
+
+    posted_async_events.pop();
+
+    for (auto& [key, value] : event) {
+      enigma_user::ds_map_add(enigma_user::async_load, key, value);
+    }
+
+    instance_event_iterator = &dummy_event_iterator;
+    for (iterator it = instance_list_first(); it; ++it) {
+      // it->myevent_asyncsteam();
```

Comment:
@RobertBColton 

Here in this line, the ``myevent_asyncsteam()`` is undefined so we need to implement it somewhere.

Reply:
This was done already, you've implemented and it works correctly.